Assets/_PlatformerDevelopment/Scripts/Player/PlayerMovementBehaviour.cs

```diff
@@ -0,0 +1,70 @@
+﻿using UnityEngine;
+using UnityEngine.InputSystem;
+
+namespace PersonalDevelopment
+{
+    [RequireComponent(typeof(PlayerInput))]
+    [RequireComponent(typeof(Rigidbody))]
+    public class PlayerMovementBehaviour : MonoBehaviour
+    {
+        // Components
+        private PlayerCharacterBindings _bindings = null;
+        private Rigidbody _rigidbody = null;
+        
+        // Movement 
+        [SerializeField] private float _movementSpeed = 5f;
+        private Vector2 Axis2D { get; set; }
+
+        #region mono
+        
+        // Start is called before the first frame update
+        private void Awake()
+        {
+            // This will never be null, since this won't even compile without RigidBody on this gameobject
+            _rigidbody = GetComponent<Rigidbody>();
+            _bindings = new PlayerCharacterBindings();
+        }
+        
+        void OnEnable()
+        {
+            _bindings.Player.Move.performed += OnMovePressed;
+            _bindings.Player.Move.canceled += OnMoveCancelled;
+            _bindings.Enable();
+        }
+
+        private void OnDisable()
+        {
+            _bindings.Player.Move.performed -= OnMovePressed;
+            _bindings.Player.Move.canceled -= OnMoveCancelled;
+            _bindings.Disable();
+        }
+
+        private void FixedUpdate()
+        {
+            Movement();
+        }
+
+        #endregion
+        
+        #region Movement
+
+        private void Movement()
+        {
+            var movement = new Vector3(Axis2D.x, 0 ,0) * _movementSpeed * Time.fixedDeltaTime;
```

Comment:
wouldnt you want to set Axis2D.y as the z-axis for the Vector3. So that you can move forward-backward?

Reply:
this is for 2d platformer so only one dimension

Reply:
Sounds good, then you can use an axis1d instead of 2d I think, since it only needs to be a 1d axis-composite (at least from the documentation). But that up to you if you want to or not

Reply:
Oh i stuck with 2D because of Jump, the up button is still used